lib/netplay/netplay.cpp

```diff
@@ -2891,6 +2891,9 @@ static void NETcheckPlayers()
 // We should not block here.
 bool NETrecvNet(NETQUEUE *queue, uint8_t *type)
 {
+	ASSERT_OR_RETURN(false, activeConnProvider != nullptr, "No active connection provider!");
+	activeConnProvider->processConnectionStateChanges();
+
```

Comment:
The `ASSERT_OR_RETURN` is problematic, as this condition is expected in at least challenge / skirmish lobbies and games, as well as when playing back a replay. Perhaps just a conditional?

```suggestion
	if (activeConnProvider)
	{
		activeConnProvider->processConnectionStateChanges();
	}

```

Also, could we safely move this below the `if (!NetPlay.bComms)` check? (Note: replays of multiplayer games would restore `NetPlay.bComms = true`, and `activeConnProvider` would be expected to be null.)

Reply:
Sure, you're right. I'll fix that.